crates/ruff_linter/src/rules/ruff/rules/missing_fstring_syntax.rs

```diff
@@ -0,0 +1,177 @@
+use memchr::memchr2_iter;
+use ruff_diagnostics::{AlwaysFixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{self as ast};
+use ruff_python_literal::format::FormatSpec;
+use ruff_python_parser::parse_expression;
+use ruff_python_semantic::SemanticModel;
+use ruff_source_file::Locator;
+use ruff_text_size::{Ranged, TextRange};
+use rustc_hash::FxHashSet;
+
+/// ## What it does
+/// Checks for strings that contain f-string syntax but are not f-strings.
+///
+/// ## Why is this bad?
+/// An f-string missing an `f` at the beginning won't format anything, and instead
+/// treat the interpolation syntax as literal.
+///
+/// Since there are many possible string literals which contain syntax similar to f-strings yet are not intended to be,
+/// this lint applies the following criteria to the string literal:
+/// 1. The string literal is not a standalone expression: it needs to be part of an assignment, function call, and so on.
+/// 2. If it's an argument in a function call that takes keyword arguments, no keywords can be used as identifiers within the formatting syntax.
+/// 3. Every identifier in a {...} section needs to be a valid, in-scope reference, and the string should have at least one identifier.
+/// 4. All format specifiers should be valid.
+///
+/// ## Example
+///
+/// ```python
+/// name = "Sarah"
+/// dayofweek = "Tuesday"
+/// msg = "Hello {name}! It is {dayofweek} today!"
+/// ```
+///
+/// It should instead be:
+/// ```python
+/// name = "Sarah"
+/// dayofweek = "Tuesday"
+/// msg = f"Hello {name}! It is {dayofweek} today!"
+/// ```
+#[violation]
+pub struct MissingFStringSyntax;
+
+impl AlwaysFixableViolation for MissingFStringSyntax {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!(r#"Possible f-string without an `f` prefix"#)
+    }
+
+    fn fix_title(&self) -> String {
+        "Add an `f` prefix to the f-string".into()
+    }
+}
+
+pub(crate) fn missing_fstring_syntax(
```

Comment:
We often do a weird thing here where we add a comment like:

```
/// RUF027
```

It just makes searching by code much easier (e.g., "Go to the definition of rule `RUF027`").